### PR TITLE
Add validation for ANALYSIS options

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
I'm looking to moving all config validations to the config package in the end.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/analysis.go [hunk 1]
```diff
@@ -18,15 +18,21 @@ import "fmt"
...
+	// Required: The expected query result.
 	Expected AnalysisExpected `json:"expected"`
-	Interval Duration         `json:"interval"`
-	// Maximum number of failed checks before the query result is considered as failure.
-	// For instance, If 1 is set, the analysis will be considered a failure after 2 failures.
+	// Required: Run a query at specified intervals.
+	Interval Duration `json:"interval"`
+	// Acceptable number of failures. For instance, If 1 is set,
+	// the analysis will be considered a failure after 2 failures.
+	// Defaults to 0.
```
Comment (nghialv) [inline pkg/config/analysis.go:31]: nit: To be unified `Default is 0.`

File: pkg/config/analysis.go [hunk 2]
```diff
@@ -18,15 +18,21 @@ import "fmt"
 
 // AnalysisMetrics contains common configurable values for deployment analysis with metrics.
 type AnalysisMetrics struct {
-	Query    string           `json:"query"`
+	// Required: The unique name of provider defined in the Piped Configuration.
```
Comment (nghialv) [inline pkg/config/analysis.go:21]: nit:
``` golang
// The unique ...
// Required field.
Provider string
```

**Comment by nakabonne** [comment]
@nghialv Thank you. Fixed nitpicks ones.

**Comment by nghialv** [comment]
Nice.
/approve

**Pull request merged**